scene/2d/screen_button.cpp

```diff
@@ -86,6 +118,15 @@ void TouchScreenButton::_notification(int p_what) {
 					draw_texture(texture,Point2());
 			}
 
+			if (!get_tree()->is_editor_hint() && !get_tree()->is_debugging_collisions_hint())
+				return;
+			if (shape.is_valid()) {
```

Comment:
if(get_tree()->is_editor_hint() && !get_tree()->is_debugging_collisions_hint() && shape.is_valid()) would be cleaner